### PR TITLE
Remove type params from RetryStrategy, but keep them in RetryStrategy…

### PR DESCRIPTION
**Pull request by sugmanue**
## Motivation and Context

As discussed during the internal review this change removes the type params from the `RetryStrategy` interface.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ private AwsRetryStrategy() {
      *
      * @return A retry strategy that does not retry.
      */
-    public static RetryStrategy<?, ?> none() {
+    public static RetryStrategy none() {
```
Comment (millems) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java:86]: doNotRetry, to match doNotBackoff?

**Review by millems — APPROVED** [review]

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java (hunk 1 above)
Comment (sugmanue) [reply core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java:86]: Sounds good, but, do we have a doNotBackoff defined anywhere? For backoff strategy the corresponding is called `retryImmediately` not `doNotBackoff`

File: core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java [hunk 2]
```diff
@@ -79,12 +79,12 @@ public static RetryStrategy addRetryConditions(RetryStrategy strategy) {
     }
 
     /**
-     * Returns a retry strategy that does not retry.
+     * Returns a retry strategy that do not retry.
```
Comment (millems) [inline core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java:82]: does*

Comment (sugmanue) [reply core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java:82]: 🤦 I will sneak in a fix on the next PR.

**Pull request merged**